components/scream/src/physics/spa/spa_functions.hpp

```diff
@@ -149,46 +150,14 @@ struct SPAFunctions
     // The weights stores the remapping weight to be applied to the source grid data for this location
     //   in the target data.
     SPAHorizInterp() = default;
-    SPAHorizInterp(const int length_)
+    SPAHorizInterp(const ekat::Comm& comm)
     {
-      length = length_;
-      weights = view_1d<Real>("",length_);
-      source_grid_loc = view_1d<Int>("",length_);
-      target_grid_loc = view_1d<Int>("",length_);
+      m_comm = comm;
     }
+    // Horizontal Remap
+    GSMap gsmap;
```

Comment:
Should we call this type `HorizontalRemap` or `HorizontalMapping` or ...?

Reply:
Great. I'll change the name.  We hadn't settled the debate in the old PR so I stuck with GSMap and made sure it would be something I could easily find/replace.  I'll use `HorizontalRemap` unless there is a strong favor for the other suggestion.